src/testing/fast_array_utils/__init__.py

```diff
@@ -37,48 +40,245 @@ def __call__(  # noqa: D102
+        """Create from qualnames of type and inner type."""
+        mod, name = qualname.rsplit(".", 1)
+        return cls(mod, name, ArrayType.from_qualname(inner) if inner else None)
+
+    def __repr__(self) -> str:  # noqa: D105
+        rv = f"{self.mod}.{self.name}"
+        return f"{rv}[{self.inner}]" if self.inner else rv
+
+    @cached_property
+    def cls(self) -> type[Array]:  # noqa: PLR0911
+        """Array class for :func:`isinstance` checks."""
+        match self.mod, self.name, self.inner:
+            case "numpy", "ndarray", None:
+                return np.ndarray
+            case "scipy.sparse", (
+                "csr_array" | "csc_array" | "csr_matrix" | "csc_matrix"
+            ) as cls_name, None:
+                import scipy.sparse
+
+                return getattr(scipy.sparse, cls_name)  # type: ignore[no-any-return]
+            case "cupy", "ndarray", None:
+                import cupy as cp
+
+                return cp.ndarray  # type: ignore[no-any-return]
+            case "cupyx.scipy.sparse", ("csr_matrix" | "csc_matrix") as cls_name, None:
+                import cupyx.scipy.sparse as cu_sparse
+
+                return getattr(cu_sparse, cls_name)  # type: ignore[no-any-return]
+            case "dask.array", cls_name, _:
+                if TYPE_CHECKING:
+                    from dask.array.core import Array as DaskArray
+                else:
+                    from dask.array import Array as DaskArray
+
+                return DaskArray
+            case "h5py", "Dataset", _:
+                import h5py
+
+                return h5py.Dataset  # type: ignore[no-any-return]
+            case "zarr", "Array", _:
+                import zarr
+
+                return zarr.Array
+            case _:
+                msg = f"Unknown array class: {self}"
+                raise ValueError(msg)
+
+    def random(
+        self,
+        shape: tuple[int, int],
+        *,
+        dtype: _DTypeLikeFloat32 | _DTypeLikeFloat64 | None,
+        gen: np.random.Generator | None = None,
+    ) -> Array:
+        """Create a random array."""
+        gen = np.random.default_rng(gen)
+
+        match self.mod, self.name, self.inner:
+            case "numpy", "ndarray", None:
+                return gen.random(shape, dtype=dtype or np.float64)
+            case "scipy.sparse", (
+                "csr_array" | "csc_array" | "csr_matrix" | "csc_matrix"
+            ) as cls_name, None:
+                fmt, container = cls_name.split("_")
+                return random_mat(shape, format=fmt, container=container, dtype=dtype)  # type: ignore[arg-type]
+            case "cupy", "ndarray", None:
+                raise NotImplementedError
+            case "cupyx.scipy.sparse", ("csr_matrix" | "csc_matrix") as cls_name, None:
+                raise NotImplementedError
+            case "dask.array", cls_name, _:
+                raise NotImplementedError
```

Comment:
Could we maybe somehow recurse here or similar to get dask random?

I'd be at the minimum curious how the benchmark plays out with dask.

Reply:
how would one do that? call `.persist()` to make sure the random data is generated ahead of time? Does it work that way?

Reply:
Why would you need the data ahead of time?  
```python
map_blocks(random_function, chunks=chunk_shape, ....)
```
Was thinking of something like this or even just `da.Array(random_array)` (or similar, not sure of the API off-hand)

Reply:
This referred to benchmarks: I don’t want to measure how fast numpy can create random data.

But sure, I can probably create random dask data by recursing!

Reply:
Would look something like this:

```py
import dask.array as da

arr = da.zeros(shape, dtype=dtype, chunks=_half_chunk_size(shape))
return arr.map_blocks(
    lambda x: self.random(x.shape, dtype=x.dtype, gen=gen), dtype=dtype
)
```